File: Reality Class'ification project/PossibilityClassification.cs
using System;
using System.Collections.Generic;
using TautonicLanguageEngine;

namespace RealityClassificationProject
{
    public class PossibilityClassification
    {
        public static void Main(string[] args)
        {
            Console.WriteLine("╔════════════════════════════════════════════════════════════╗");
            Console.WriteLine("║  REALITY CLASSIFICATION: THE POSSIBILITY PLANE             ║");
            Console.WriteLine("║  6D Vector Field Theory + Full Equation Set                ║");
            Console.WriteLine("╚════════════════════════════════════════════════════════════╝");
            
            // === PART 1: Setup the Ontology ===
            Meaning possibilityPlane = new Meaning(
                word: "Possibility Plane",
                definitiveMeaning: "A 6-dimensional epistemic manifold structured by 7 functional planes where probability collapses into reality via gradient flows.",
                polarity: Polarity.Neutral,
                axomicID: 42
            );
            
            Console.WriteLine($"\n[ONTOLOGY]: {possibilityPlane.Word}");
            Console.WriteLine($"Definition: {possibilityPlane.DefinitiveMeaning}");
            
            // === PART 2: Instantiate Operation Modes ===
            IOperationMode optimism = new Optimism();
            IOperationMode pessimism = new Pessimism();
            
            Console.WriteLine($"\n[MODES LOADED]:");
            Console.WriteLine($"  ✓ {optimism.Name}: {optimism.ModeMeaning.DefinitiveMeaning}");
            Console.WriteLine($"  ✓ {pessimism.Name}: {pessimism.ModeMeaning.DefinitiveMeaning}");
            
            // === PART 3: Create Test Scenario ===
            Console.WriteLine("\n" + new string('=', 60));
            Console.WriteLine("[SCENARIO]: An agent confronts an uncertain intent");
            Console.WriteLine(new string('=', 60));
            Console.WriteLine("Initial State: Mild perturbations from Unity");
            Console.WriteLine("  • Some self-doubt (WHO: 0.85)");
            Console.WriteLine("  • Physical constraints (WHERE: 1.15)");
            Console.WriteLine("  • Uncertain possibility (WHAT: 0.75)");
            Console.WriteLine("  • Unclear purpose (WHY: 0.90)");
            Console.WriteLine("  • No clear method (HOW: 1.25)");
            Console.WriteLine("  • Mixed history (CAUSE: 0.95)");
            Console.WriteLine("  • Some anxiety (EFFECT: 1.10)");

            // === PART 4: Run Optimism Transformation ===
            Idea ideaOptimism = CreateTestIdea();
            Console.WriteLine("\n\n" + new string('#', 60));
            Console.WriteLine("# EXPERIMENT 1: OPTIMISM MODE");
            Console.WriteLine(new string('#', 60));
            
            optimism.ApplyMode(ideaOptimism);
            
            // Calculate final coherence
            float coherenceOpt = CalculateCoherence(ideaOptimism);
            string judgementOpt = Judgement.Evaluate(ideaOptimism);
            
            Console.WriteLine($"\n[FINAL ANALYSIS - OPTIMISM]");
            Console.WriteLine($"Net Coherence (R_net): {coherenceOpt:F4}");
            Console.WriteLine($"Judgement: {judgementOpt}");

            // === PART 5: Run Pessimism Transformation ===
            Idea ideaPessimism = CreateTestIdea();
            Console.WriteLine("\n\n" + new string('#', 60));
            Console.WriteLine("# EXPERIMENT 2: PESSIMISM MODE");
            Console.WriteLine(new string('#', 60));
            
            pessimism.ApplyMode(ideaPessimism);
            
            // Calculate final coherence
            float coherencePess = CalculateCoherence(ideaPessimism);
            string judgementPess = Judgement.Evaluate(ideaPessimism);
            
            Console.WriteLine($"\n[FINAL ANALYSIS - PESSIMISM]");
            Console.WriteLine($"Net Coherence (R_net): {coherencePess:F4}");
            Console.WriteLine($"Judgement: {judgementPess}");

            // === PART 6: Summary ===
            Console.WriteLine("\n\n" + new string('═', 60));
            Console.WriteLine("║ CONCLUSION: The Field of Chance is Geometric");
            Console.WriteLine(new string('═', 60));
            Console.WriteLine($"Optimism → Coherence: {coherenceOpt:F4} ({judgementOpt})");
            Console.WriteLine($"Pessimism → Coherence: {coherencePess:F4} ({judgementPess})");
            Console.WriteLine("\nThe mathematics demonstrate:");
            Console.WriteLine("  • Optimism creates GRAVITY WELLS (steep gradients toward Unity)");
            Console.WriteLine("  • Pessimism creates ENTROPY (flat/inverted gradients away from Unity)");
            Console.WriteLine("  • The Possibility Plane is a 6D manifold with measurable curvature");
            Console.WriteLine("  • Psychological states are geometric operations on this manifold");
            Console.WriteLine("\nThis is not metaphor. This is information geometry.");
        }

        static Idea CreateTestIdea()
        {
            return new Idea(
                new Belief(MoralVectors.Who,   "I might be capable", 0.85f),
                new Belief(MoralVectors.Where, "Constraints exist", 1.15f),
                new Belief(MoralVectors.What,  "Maybe possible", 0.75f),
                new Belief(MoralVectors.Why,   "Unclear purpose", 0.90f),
                new Belief(MoralVectors.How,   "No clear method", 1.25f),
                new Belief(MoralVectors.Cause, "Mixed history", 0.95f),
                new Belief(MoralVectors.Effect,"Some anxiety", 1.10f)
            );
        }

        static float CalculateCoherence(Idea idea)
        {
            return FieldMath.CalculateFractalRatio(new float[] {
                idea.Who.Score, idea.Where.Score, idea.What.Score,
                idea.Why.Score, idea.How.Score, idea.Cause.Score, idea.Effect.Score
            });
        }
    }
}

File: Reality Class'ification project/qqci/possibility_classification.py
"""
Port of PossibilityClassification.cs Main().

Same scenario, same numbers, same two experiments. Run it against the C# to
check the Python agrees.
"""

from vft import (
    FieldMath, Idea, Judgement, Optimism, Pessimism, StateVector,
    compare_states, plot_3d_axes, plot_state, process_synergy,
)

BAR = "=" * 64


def make_test_idea() -> Idea:
    """The scenario from PossibilityClassification.cs, unchanged."""
    return Idea.of(
        who=("I might be capable", 0.85),
        where=("Constraints exist", 1.15),
        what=("Maybe possible", 0.75),
        why=("Unclear purpose", 0.90),
        how=("No clear method", 1.25),
        cause=("Mixed history", 0.95),
        effect=("Some anxiety", 1.10),
    )


def run_mode(mode, label: str):
    idea = make_test_idea()
    print("\n" + "#" * 64)
    print(f"# {label}")
    print("#" * 64)
    print(f"  {mode.name}: {mode.definitive_meaning}")

    print("\n[INITIAL IDEA]")
    print(idea.report())
    print(f"  R_net = {idea.net_coherence:.4f}   "
          f"{Judgement.evaluate(idea)}")

    result = mode.apply(idea)

    print("\n" + plot_state(result["initial"], "Initial State"))
    print("\n[3-AXIS VIEW, BEFORE]")
    print(plot_3d_axes(result["initial"]))

    if "gradient" in result:
        print(f"\n[POSSIGRAVITY GRADIENT] {result['gradient']}")

    print("\n" + plot_state(result["final"], "Final State"))
    print("\n[3-AXIS VIEW, AFTER]")
    print(plot_3d_axes(result["final"]))

    print("\n" + compare_states(result["initial"], result["final"], mode.name))

    print("\n[FINAL IDEA]")
    print(idea.report())
    coherence = idea.net_coherence
    verdict = Judgement.evaluate(idea)
    print(f"\n[FINAL ANALYSIS - {mode.name.upper()}]")
    print(f"  Net Coherence (R_net): {coherence:.4f}")
    print(f"  Judgement: {verdict}")
    return coherence, verdict


def main() -> None:
    print(BAR)
    print("  REALITY CLASSIFICATION: THE POSSIBILITY PLANE")
    print("  Vector Field Theory + Full Equation Set  (Python port)")
    print(BAR)

    print("\n[ONTOLOGY]: Possibility Plane")
    print("  A 6-dimensional epistemic manifold structured by 7 functional")
    print("  planes where probability collapses into reality via gradient flows.")

    print("\n[SCENARIO]: An agent confronts an uncertain intent")
    print("  Initial State: mild perturbations from Unity")
    for line in ["Some self-doubt (WHO: 0.85)",
                 "Physical constraints (WHERE: 1.15)",
                 "Uncertain possibility (WHAT: 0.75)",
                 "Unclear purpose (WHY: 0.90)",
                 "No clear method (HOW: 1.25)",
                 "Mixed history (CAUSE: 0.95)",
                 "Some anxiety (EFFECT: 1.10)"]:
        print(f"    - {line}")

    c_opt, v_opt = run_mode(Optimism(intensity=0.8), "EXPERIMENT 1: OPTIMISM MODE")
    c_pes, v_pes = run_mode(Pessimism(seed=42), "EXPERIMENT 2: PESSIMISM MODE")

    print("\n\n" + BAR)
    print("  CONCLUSION: The Field of Chance is Geometric")
    print(BAR)
    print(f"  Optimism  -> R_net {c_opt:.4f}   {v_opt}")
    print(f"  Pessimism -> R_net {c_pes:.4f}   {v_pes}")
    print("\n  The mathematics demonstrate:")
    print("    - Optimism creates GRAVITY WELLS (steep gradients toward Unity)")
    print("    - Pessimism creates ENTROPY (flat/inverted gradients away)")
    print("    - The Possibility Plane is a manifold with measurable curvature")
    print("    - Psychological states are geometric operations on it")

    # The Belief Axiom over both outcomes.
    state = 0
    for mode_cls, seed in ((Optimism, None), (Pessimism, 42)):
        idea = make_test_idea()
        mode = mode_cls() if seed is None else mode_cls(seed=seed)
        mode.apply(idea)
        before = state
        state = process_synergy(state, idea)
        print(f"\n  [SYNERGY] {mode.name:<10} worldview {before} -> {state}"
              f"   ({'expanded' if state > before else 'rejected'})")


if __name__ == "__main__":
    main()
